### PR TITLE
detect/bytemath: Handle out-of-order keywords

### PR DESCRIPTION
**Pull request by jlucovsky**
Issue: 5077

This commit allows byte_math keywords past <rvalue> to be in arbitrary
order to provide better compatibility with Snort's handling.

"Arbitrary order" refers to an order that is different from Snort's documented keyword order.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [5077](https://redmine.openinfosecfoundation.org/issues/5077)

Describe changes:
- Reworked PCRE capture to allow arbitrary order of keywords.
- Added unittest with out of order keywords.

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/detect-bytemath.c [hunk 1]
```diff
@@ -263,6 +267,7 @@ static DetectByteMathData *DetectByteMathParse(DetectEngineCtx *de_ctx, const ch
     char tmp_str[128] = "";
 
     ret = DetectParsePcreExec(&parse_regex, arg, 0, 0);
+    SCLogNotice("byte_math expression: \"%s\"; ret: %d", arg, ret);
```
Comment (victorjulien) [inline src/detect-bytemath.c:270]: debug

**Review by victorjulien — COMMENTED** [review]
wondering if we should get rid of pcre for parsing completely, like we're doing with xbits

File: src/detect-bytemath.c (hunk 1 above)
Comment (jlucovsky) [reply src/detect-bytemath.c:270]: Will fix.

**Comment by jlucovsky** [comment]
> wondering if we should get rid of pcre for parsing completely, like we're doing with xbits

Is that in progress? https://github.com/OISF/suricata/blob/master/src/detect-xbits.c#L206

**Comment by catenacyber** [comment]
> > wondering if we should get rid of pcre for parsing completely, like we're doing with xbits
> 
> Is removal of pcre from xbits work in progress? https://github.com/OISF/suricata/blob/master/src/detect-xbits.c#L206

I think there are PRs from Shivani...

**Comment by jlucovsky** [comment]
Continued in #7113

**Pull request closed without merging**